ios/brave-ios/Sources/Brave/Frontend/UserContent/UserScripts/Scripts_Dynamic/Scripts/Sandboxed/SelectorsPollerScript.js

```diff
@@ -1070,19 +1084,96 @@ window.__firefox__.execute(function($) {
       unhideSelectorsMatchingElementsAndTheirParents(nodesWithExtractedURLs)
     }
 
-    // 4. Set the rules on the stylesheet. So far we don't have any rules set
+    // 4. Perform procedural actions before setting stylesheet
+    if (CC.hasProceduralActions) {
+      executeProceduralActions()
+    }
+
+    // 5. Set the rules on the stylesheet. So far we don't have any rules set
     // We could do this earlier but it causes elements to hide and unhide
     // It's best to wait to this period so we have
     setRulesOnStylesheet()
-    // 5. Start our queue pump if we need to
+    // 6. Start our queue pump if we need to
     scheduleQueuePump(args.hideFirstPartyContent, args.genericHide)
 
     if (!args.hideFirstPartyContent) {
-      // 6. Start listening to new urls
+      // 7. Start listening to new urls
       startURLMutationObserver()
     }
   }
 
+  /// Process any :style, :remove, :remove-attr, or :remove-class selectors
+  const executeProceduralActions = (added) => {
+    // If passed a list of added elements, do not query the entire document
+    if (CC.proceduralActionFilters === undefined) {
+      return
+    }
+
+    /**
+    * Adds given styles to allStyleRules
+    * @param {Object} entry The entry to add
+    */
+    const addStyleRule = (entry) => {
+      const rule = entry.selector + '{' + entry.rules.join(';') + ';}'
+      CC.allStyleRules.push(rule)
+    }
+
+    const performAction = (selector, element, action) => {
+      console.debug('Performing action', action, 'on element', element);
+      if (action === undefined) {
+        addStyleRule({ selector, rules: ['display: none !important'] })
+      } else if (action.type === 'style') {
+        addStyleRule({ selector, rules: [action.arg] })
+      } else if (action.type === 'remove') {
+        element.remove()
+      } else if (action.type === 'remove-attr') {
+        // We can remove attributes without checking if they exist
+        element.removeAttribute(action.arg)
+      } else if (action.type === 'remove-class') {
+        // Check if the element has any classes to remove because
+        // classList.remove(tokens...) always triggers another mutation
+        // even if nothing was removed.
+        if (element.classList.contains(action.arg)) {
+          element.classList.remove(action.arg);
+        }
+      }
+    }
+    for (const { selector, action } of CC.proceduralActionFilters) {
+      console.debug('Processing procedural selector', selector, action);
```

Comment:
probably don't want to spam the console with these; there could be a lot on e.g. sites with infinite scrolling feeds